shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -303,14 +324,22 @@ class JSBackend(allowUnresolvedSymbols: Boolean) {
         case Var("string") =>
           // JS is dumb so `instanceof String` won't actually work on "primitive" strings...
           JSBinary("===", scrut.member("constructor"), JSLit("String"))
-        case Var(name) => topLevelScope.getType(name) match {
-          case S(ClassSymbol(_, runtimeName, _, _, _)) => JSInstanceOf(scrut, JSIdent(runtimeName))
-          case S(NewClassSymbol(lexicalName, _, _, _, _, _)) => JSInstanceOf(scrut, JSMember(JSIdent(lexicalName), JSIdent(JSLit.makeStringLiteral("class"))))
-          case S(ModuleSymbol(lexicalName, _, _, _, _, _)) => JSInstanceOf(scrut, JSMember(JSIdent(lexicalName), JSIdent(JSLit.makeStringLiteral("class"))))
-          case S(TraitSymbol(_, runtimeName, _, _, _)) => JSIdent(runtimeName)("is")(scrut)
-          case S(_: TypeAliasSymbol) => throw new CodeGenError(s"cannot match type alias $name")
-          case S(_: MixinSymbol) => throw new CodeGenError(s"cannot match mixin $name")
-          case N => throw new CodeGenError(s"unknown match case: $name")
+        case Var(name) => scope.resolveValue(name) match {
+          case S(NewClassSymbol(lexicalName, _, _, _, _, _, _, _)) =>
```

Comment:
```suggestion
          case S(sym: NewClassSymbol) =>
```
Same below. When most fields are `_`, just match on the type.